### ISSUE TITLE
Uncaught TypeError: Cannot read property 'types' of undefined

### ISSUE THREAD

**Issue by smit9234** [4teamwork/cypress-drag-drop#9]
I am getting the following error when the dragstart method is called

![image](https://user-images.githubusercontent.com/7388782/55260360-6c8da300-5225-11e9-9140-62052da6d6d6.png)

Uncaught TypeError: Cannot read property 'types' of undefined

This error originated from your application code, not from Cypress.

When Cypress detects uncaught errors originating from your application it will automatically fail the current test.

This behavior is configurable, and you can choose to turn this off by listening to the 'uncaught:exception' event.

https://on.cypress.io/uncaught-exception-from-application

I have the require('cypress-drag-drop) in my command.js file.  I am not sure how else to debug this error.  Thoughts?  I am using Cypress 3.2 and Chrome version 73.

**Comment by smit9234** [comment]
Are you asking for my cypress or the application code?  I am using the syntax provided in the example.  The error occurs on the trigger. Cypress console isn't showing an relevant stack trace.  Debugging doesn't show any other errors.

Here is the line of code:

    cy.get('#dragHazardCardIconAdd').eq(0).drag('#dropNoHazards');

**Comment by bierik** [comment]
@smit9234 Can you give me a reproduction link (I'm fine with https://codepen.io/ or https://codesandbox.io) or a link to a repo so I can reproduce your issue. I guess there is something in your application that tries to get `types` of an undefined object when dragging.

**Comment by technomage** [comment]
This comes from the dragstart not haing the payload (dataTransfer) data expected in an HTML5 drag/drop.

**Comment by bierik** [comment]
@smit9234 So can you try using the newest version https://github.com/4teamwork/cypress-drag-drop/releases/tag/v1.2.1. Because there is now a default dataTransfer object since https://github.com/4teamwork/cypress-drag-drop/pull/10.

**Comment by smit9234** [comment]
@bierik 
Sorry I am not able to give you the source code (intranet app). 

I tried using version 1.2.1, and now I get this error:
`TypeError:` Cannot read property 'id' of undefined

**Comment by bierik** [comment]
@smit9234 Can you isolate the code that is producing the error and provide me a reproduction link?

**Comment by technomage** [comment]
@bierik Is the "default" data transfer going to have the data that would be present in the real drag/drop?  Ideally the contents would be populated by the actual components that support drag/drop.

**Comment by smit9234** [comment]
@bierik 
I am not able to provide our source code. However I did try to make the plugin work against the following site that is an demo for dropping and dragging: 

https://php.quicoto.com/examples/25/

Here is my cypress test:

    describe('drop and drag box 1', () => {
      it('should drag box 1 and drop in the larger box', () => {
      cy.visit('https://php.quicoto.com/examples/25/');
      cy.get('div[data-title="Box 1"]').drag('#drop');
      });
    });

This gives me the following error:
Uncaught TypeError: Cannot read property 'setData' of undefined

Does the plugin in work with any type of drag of drop javascript?

**Comment by baronvonbirra** [comment]
Exactly the same issue happening here.
Here's my code:
```
    cy.get(`[data-test="recordmenu:items"]`)
      .find('.item')
      .eq(0)
      .drag('.add-page');
```
Here the error:
<img width="413" alt="image" src="https://user-images.githubusercontent.com/18036304/58952936-5eb04d00-8795-11e9-8378-6c343361a78e.png">

Any solutions? What could be wrong here? 

Thanks!

**Comment by bierik** [comment]
@smit9234 I released a new version 1.3.0 which includes the fix for the datatransfer object. I tried your example on https://php.quicoto.com/examples/25/ and it worked:

![Kapture 2019-09-10 at 19 28 04](https://user-images.githubusercontent.com/1637820/64636029-2d34c680-d401-11e9-907e-d32d653773c4.gif)

I hoped this helps.
